flink_jobs_v2/Timelines/src/main/java/timelines/Timeline.java

```diff
@@ -461,74 +459,64 @@ public int opInt(int[][][] truthTable, int op, int a, int b) {
 
         return result;
     }
-
     /**
-     * aggregates a timeline with a timeline of downtime statuses, defined from
-     * the downtime data source
      *
-     * @param downtimeline the timeline containing the defined downtime
-     * timestamps
-     * @param downDefaultInt the downtime status
+     * Fills a timeline with a given status for specific periods, replacing status of timestamp's in the period with the given status 
+     * @param periods, a list of String[] that contains startDate, endDate that
+     * the status appears
+     * @param intStatus, the status to fill the timeline for the specific
+     * periods
+     * @param runDate, the timeline date
+     *
      */
-    public void aggregateDownTime(Timeline downtimeline, int downDefaultInt) {
-        if (downtimeline.isEmpty()) {
-            return;
-        }
-        if (this.isEmpty()) {
-            this.bulkInsert(downtimeline.getSamples());
-            // Optimize even when we have a single timeline for aggregation
-            this.optimize();
-            return;
-        }
-
-        Timeline result = new Timeline();
-
-        // Slice for first
-        for (DateTime point : this.getPoints()) {
-            result.insert(point, -1);
-        }
-        // Slice for second 
-        for (DateTime point : downtimeline.getPoints()) {
-            result.insert(point, -1);
-        }
+    public void fillTimelineWithStatus(ArrayList<String[]> periods, Integer intStatus, String runDate) throws ParseException {
```

Comment:
Why are we using an array list for accepting three standard arguments such as a start_date, end_date and status?

Reply:
I guess it's used to try at once to fill a timeline with different statuses at different parts? It might be more clean to keep the function simpler with the capability to just be able to fill a specific period of the timeline with one specific status. And if it is needed call the function multiple times. Since the object that this method belongs to is a Timeline of statuses it may be worth considering naming the method `fill()` or `fillWithStatus()`

Reply:
Yes it was designed as such in order to update the timeline with a status for all the periods of time it is defined to appear. changed to accept one period and change status during this time